manifests/init.pp

```diff
@@ -0,0 +1,166 @@
+  Integer[0,2]                $mirrors_mode                      = 0,
+  Optional[Array[String[1]]]  $missing_logfiles                  = undef,
+  Optional[String[1]]         $modules_dir                       = undef,
+  Optional[String[1]]         $os_version_file                   = undef,
+  Optional[String[1]]         $password_file                     = undef, #should keep? suggestion: /etc/shadow
+  Integer[0,1]                $phalanx2_dirtest                  = 0,
+  Optional[Array[String[1]]]  $pkgmgr_no_vrfy                    = undef,
+  String[1]                   $pkgmgr                            = 'RPM',
+  Optional[Array[String[1]]]  $port_path_whitelist               = undef,
+  Optional[Array[String[1]]]  $port_whitelist                    = undef,
+  Optional[Array[String[1]]]  $pwdless_accounts                  = undef,
+  Optional[String[1]]         $readlink_cmd                      = undef,
+  Integer[0,1]                $rotate_mirrors                    = 1,
+  Optional[Array[String[1]]]  $rtkt_dir_whitelist                = undef,
+  Optional[Array[String[1]]]  $rtkt_file_whitelist               = undef,
+  Enum['THOROUGH','LAZY']     $scan_mode_dev                     = 'THOROUGH',
+  Optional[String[1]]         $scriptdir                         = undef,
+  Optional[Array[String[1]]]  $scriptwhitelist                   = undef,
+  Optional[Array[String[1]]]  $shared_lib_whitelist              = undef,
+  Integer[0,1]                $skip_inode_check                  = 0,
+  Optional[String[1]]         $ssh_config_dir                    = undef,
+  Optional[Array[String[1]]]  $startup_paths                     = undef, #suggestion: /etc/rc.d /etc/rc.local ??
+  Optional[String[1]]         $stat_cmd                          = undef,
+  Optional[Array[String[1]]]  $suspscan_dirs                     = undef,
+  Integer                     $suspscan_maxsize                  = 1024000,
+  String[1]                   $suspscan_temp                     = '/dev/shm',
+  Integer                     $suspscan_thresh                   = 200,
+  Optional[Array[String[1]]]  $suspscan_whitelist                = undef,
+  Optional[Array[String[1]]]  $syslog_config_file                = undef,
+  Optional[String[1]]         $tmpdir                            = undef,
+  Optional[Array[String[1]]]  $uid0_accounts                     = undef,
+  Optional[Array[String[1]]]  $update_lang                       = undef,
+  Integer[0,1]                $update_mirrors                    = 1,
+  Integer[0,1]                $updt_on_os_change                 = 0,
+  Integer[0,1]                $use_locking                       = 1,
+  Optional[Array[String[1]]]  $user_fileprop_files_dirs          = undef,
+  Integer[0,1]                $use_sunsum                        = 0,
```

Comment:
If parameters are not set by default in the configuration file they generally should not be set by default here either and should be marked as `Optional`.

Reply:
it is the default...

> #
> # If the 'SOLARIS' package manager is used, then it is possible to use the
> # checksum (hash) value stored for a file. However, this is only a 16-bit
> # checksum, and as such is not nearly as secure as, for example, a SHA-2 value.
> # If the option is set to '0', then the checksum is not used and the hash
> # function given by HASH_CMD is used instead. To enable this option, set its
> # value to '1'. The Solaris 'sum' command must be present on the system if this
> # option is used.
> #
> # The default value is '0'.
> #
> #USE_SUNSUM=0